core/optaplanner-core-impl/src/main/java/org/optaplanner/core/impl/score/director/incremental/IncrementalScoreDirector.java

```diff
@@ -175,24 +175,24 @@ public void afterElementAdded(ListVariableDescriptor<Solution_> variableDescript
     }
 
     @Override
-    public void beforeElementMoved(ListVariableDescriptor<Solution_> variableDescriptor,
-            Object sourceEntity, int sourceIndex, Object destinationEntity, int destinationIndex) {
+    public void beforeElementRemoved(ListVariableDescriptor<Solution_> variableDescriptor, Object entity, int index) {
         throw new UnsupportedOperationException("Not yet supported.");
     }
 
     @Override
-    public void afterElementMoved(ListVariableDescriptor<Solution_> variableDescriptor,
-            Object sourceEntity, int sourceIndex, Object destinationEntity, int destinationIndex) {
+    public void afterElementRemoved(ListVariableDescriptor<Solution_> variableDescriptor, Object entity, int index) {
         throw new UnsupportedOperationException("Not yet supported.");
     }
 
     @Override
-    public void beforeElementRemoved(ListVariableDescriptor<Solution_> variableDescriptor, Object entity, int index) {
+    public void beforeSubListChanged(ListVariableDescriptor<Solution_> variableDescriptor, Object entity, int fromIndex,
+            int toIndex) {
         throw new UnsupportedOperationException("Not yet supported.");
     }
 
     @Override
-    public void afterElementRemoved(ListVariableDescriptor<Solution_> variableDescriptor, Object entity, int index) {
+    public void afterSubListChanged(ListVariableDescriptor<Solution_> variableDescriptor, Object entity, int fromIndex,
```

Comment:
Naming must align to ListVariableListener interface, so maybe before/afterListVariableChanged(...)